eth-custodian/contracts/EthCustodianProxy.sol

```diff
@@ -0,0 +1,128 @@
+        ethCustodianImpl.depositToNear{value: msg.value}(nearRecipientAccountId, 0);
+    }
+
+    function depositToEVM(
+        string memory ethRecipientOnNear
+    ) external payable whenNotPaused(PAUSED_DEPOSIT_TO_EVM) {
+        ethCustodianImpl.depositToEVM{value: msg.value}(ethRecipientOnNear, 0);
+    }
+
+    function withdraw(
+        bytes calldata proofData,
+        uint64 proofBlockHeight
+    ) external {
+        if (proofBlockHeight > migrationBlockHeight) {
+            _requireNotPaused(PAUSED_WITHDRAW);
+            ethCustodianImpl.withdraw(proofData, proofBlockHeight);
+        } else {
+            _requireNotPaused(PAUSED_WITHDRAW_PRE_MIGRATION);
+            bytes memory postMigrationProducer = ethCustodianImpl.nearProofProducerAccount_();
+            writeProofProducerSlot(preMigrationProducerAccount);
+            ethCustodianImpl.withdraw(proofData, proofBlockHeight);
+            writeProofProducerSlot(postMigrationProducer);
+        }
+    }
+
+    function migrateToNewProofProducer(
+        bytes calldata newProducerAccount,
+        uint64 migrationBlockNumber
+    ) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        if (keccak256(preMigrationProducerAccount) != keccak256(hex"")) {
+            revert AlreadyMigrated();
+        }
+
+        // Needs to fit in one slot
+        if (newProducerAccount.length > 31) {
+            revert ProducerAccountIdTooLong(newProducerAccount);
+        }
+
+        migrationBlockHeight = migrationBlockNumber;
+        preMigrationProducerAccount = ethCustodianImpl.nearProofProducerAccount_();
+        writeProofProducerSlot(newProducerAccount);
+    }
+
+    function pauseAll() external onlyRole(PAUSABLE_ADMIN_ROLE) {
+        uint flags = PAUSED_DEPOSIT_TO_EVM |
+            PAUSED_DEPOSIT_TO_NEAR |
+            PAUSED_WITHDRAW |
+            PAUSED_WITHDRAW_PRE_MIGRATION;
+        ethCustodianImpl.adminPause(flags);
+        _pause(flags);
+    }
+
+    function pauseImpl(uint flags) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        ethCustodianImpl.adminPause(flags);
+    }
+
+    function pauseProxy(uint flags) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        _pause(flags);
+    }
+
+    /**
+     * @dev Internal function called by the proxy contract to authorize an upgrade to a new implementation address
+     * using the UUPS proxy upgrade pattern. Overrides the default `_authorizeUpgrade` function from the `UUPSUpgradeable` contract.
+     * This function does not need to perform any extra authorization checks other than restricting the execution of the function to the admin and reverting otherwise.
+     * @param newImplementation Address of the new implementation contract.
+     * Requirements:
+     * - The caller must have the `DEFAULT_ADMIN_ROLE`.
+     */
+    function _authorizeUpgrade(
+        address newImplementation
+    ) internal override onlyRole(DEFAULT_ADMIN_ROLE) {}
+
+    function writeProofProducerSlot(bytes memory proofProducer) private {
```

Comment:
Please add an underscore prefix to the private methods.

Reply:
Done ec064b591605ef4aa0f1a08004c615b2504927dd